sdk/core/core-rest-pipeline/test/defaultLogPolicy.spec.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "chai";
+import { createPipelineFromOptions } from "../src/createPipelineFromOptions";
+import { PipelinePolicy } from "../src/pipeline";
+import { createPipelineRequest } from "../src/pipelineRequest";
+import { createHttpHeaders } from "../src/httpHeaders";
+import sinon from "sinon";
+import { DEFAULT_RETRY_POLICY_COUNT } from "../src/constants";
+
+describe("defaultLogPolicy", function () {
+  it("should be invoked on every retry", async function () {
+    const request = createPipelineRequest({
+      url: "https://bing.com",
+    });
+
+    let testSignPolicy: PipelinePolicy = {
+      name: "testSignPolicy",
+      sendRequest: async function (request, next) {
+        let response = await next(request);
+        return response;
+      },
+    };
+
+    let pipeline = createPipelineFromOptions({
+      retryOptions: { maxRetryDelayInMs: 0 },
+    });
+    pipeline.addPolicy(testSignPolicy, { phase: "Sign" });
+
+    let orderedPolicies = pipeline.getOrderedPolicies();
+    assert.deepEqual(
+      orderedPolicies.map((policy) => policy.name),
+      [
+        "proxyPolicy",
+        "decompressResponsePolicy",
+        "formDataPolicy",
+        "userAgentPolicy",
+        "setClientRequestIdPolicy",
+        "defaultRetryPolicy",
+        "tracingPolicy",
+        "redirectPolicy",
+        "testSignPolicy",
+        "logPolicy",
+      ]
+    );
+
+    let order: string[] = [];
+    orderedPolicies.map((policy) => {
+      let wrappedMethod = policy.sendRequest;
+      sinon.stub(policy, "sendRequest").callsFake(async function (request, next) {
+        order.push(policy.name);
+        return wrappedMethod(request, next);
+      });
```

Comment:
style nit, I believe sinon stores this for you now:

```suggestion
      const stub = sinon.stub(policy, "sendRequest").callsFake(async function (request, next) {
        order.push(policy.name);
        return stub.wrappedMethod(request, next);
      });
```

Reply:
That's neat. I didn't realize closures could access the object that stores them. It looks strange, but it makes sense. Done!